HW5.py

```diff
@@ -0,0 +1,46 @@
+"""
+1) Реализовать класс который бы производил разные операции над 2мя переданными числами.
+
+Пример Использования:
+
+proccess_input = ProcessInput(a=20, b=10)
+print(proccess_input.add()). # выведет 30
+print(proccess_input.subtract()). # выведет 10
+print(proccess_input.multiple()). # выведет 200
+print(proccess_input.divide()). # выведет 2
+"""
+
+
+class ProcessInput:
+    """
+    the class to perform some actions with two input digits
+    """
+
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+
+    def add(self):
+        return self.a + self.b
+
+    def subtract(self):
+        return self.a - self.b
+
+    def multiple(self):
+        return self.a * self.b
+
+    def divide(self):
+        return self.a / self.b
```

Comment:
would be nice to have try-except block, because self.b could be 0

Reply:
I have made it like this
```
    def divide(self):
        try:
            return self.a / self.b
        except ZeroDivisionError:
            print("Fortunately or unfortunately, but you cannot devide by a zero")
```
But after I decided to avoid another possible error: wrong type of input data
I have read a bit about annotation and wrote something like:
```
from typing import Union
def __init__(self, a: Union[int, float], b: Union[int, float]):
```
and was very surprised when self.a = 'a' did't rise an TypeError error.
**QUESTION#1: why and how to do it right ? ("to catch" wrong type issue in the very beginning)**
After I have to add  `except TypeError ` to every method and noticed another thing:
for both 
```
    def divide(self):
        try:
            return self.a / self.b
        except ZeroDivisionError:
            print("Fortunately or unfortunately, but you cannot devide by a zero")
        except TypeError:
            print("Cannot devide: some of your input data belong to a wrong type (not int nor float)")
```
and
```
    def divide_v2(self):
        try:
            return int(self.a / self.b)
        except TypeError:
            print("Cannot devide: some of your input data belong to a wrong type (not int nor float)")
        except ZeroDivisionError:
            print("Fortunately or unfortunately, but you cannot devide by a zero")  
```
**QUESTION#2 TypeError happens before ZeroDivisionError?**

Reply:
The first question was answered during the lesson. But the second is not because I've forgotten to ask it 🤣